generator/config/expressions.php

```diff
@@ -45,6 +46,11 @@
 $expressions = [];
 $resolvesToInterfaces = [];
 foreach ($bsonTypes as $name => $acceptedTypes) {
+    // an expression can be a string with a $-prefixed field name
+    if (! in_array('string', $acceptedTypes)) {
+        $acceptedTypes[] = 'string';
```

Comment:
Assuming this is the only code change in this PR, what about this change allows $-prefixed strings specifically? It looks like you're just accepting _any_ kind of string whenever an expression is expected.

Side note: I couldn't confirm if this was really the only code change in the PR because GitHub's PR view is not correctly labeling all generated files. In the future it'd be preferable to segregate changes to generated files into their own commit.

Reply:
I added validation of `$string`, but that's something the server would do.

> Failed to optimize pipeline :: caused by :: $abs only supports numeric types, not string